packages/mui-material/src/Divider/Divider.js

```diff
@@ -54,100 +55,136 @@ const DividerRoot = styled('div', {
         styles.textAlignLeft,
     ];
   },
-})(
-  ({ theme, ownerState }) => ({
-    margin: 0, // Reset browser default style.
-    flexShrink: 0,
-    borderWidth: 0,
-    borderStyle: 'solid',
-    borderColor: (theme.vars || theme).palette.divider,
-    borderBottomWidth: 'thin',
-    ...(ownerState.absolute && {
-      position: 'absolute',
-      bottom: 0,
-      left: 0,
-      width: '100%',
-    }),
-    ...(ownerState.light && {
-      borderColor: theme.vars
-        ? `rgba(${theme.vars.palette.dividerChannel} / 0.08)`
-        : alpha(theme.palette.divider, 0.08),
-    }),
-    ...(ownerState.variant === 'inset' && {
-      marginLeft: 72,
-    }),
-    ...(ownerState.variant === 'middle' &&
-      ownerState.orientation === 'horizontal' && {
+})(({ theme }) => ({
+  margin: 0, // Reset browser default style.
+  flexShrink: 0,
+  borderWidth: 0,
+  borderStyle: 'solid',
+  borderColor: (theme.vars || theme).palette.divider,
+  borderBottomWidth: 'thin',
+  variants: [
+    {
+      props: {
+        absolute: true,
+      },
+      style: {
+        position: 'absolute',
+        bottom: 0,
+        left: 0,
+        width: '100%',
+      },
+    },
+    {
+      props: {
+        light: true,
+      },
+      style: {
+        borderColor: theme.vars
+          ? `rgba(${theme.vars.palette.dividerChannel} / 0.08)`
+          : alpha(theme.palette.divider, 0.08),
+      },
+    },
+    {
+      props: {
+        variant: 'inset',
+      },
+      style: {
+        marginLeft: 72,
+      },
+    },
+    {
+      props: ({ ownerState }) =>
+        ownerState.variant === 'middle' && ownerState.orientation === 'horizontal',
```

Comment:
```suggestion
      props: { variant: 'middle', orientation: 'horizontal' },
```

I think object comparison should be used if possible.